app/connect_wizard.py

```diff
@@ -155,7 +156,7 @@ def Draw(self, evt):
                 self.option = dict()
                 saved_opts = get_saved_connection(self.engine.name)
                 for opt in self.engine.required_opts:
-                    if opt[0] in saved_opts.keys():
+                    if opt[0] in list(saved_opts.keys()):
```

Comment:
This applies to many other cases of `for something in list(dict.keys()):` and `if something in list(dict.keys()):`, so I won't comment on them all.